src/Sync.ts

```diff
@@ -0,0 +1,355 @@
+    }
+    spliceProgress.stop();
+    return files;
+  }
+  async getSplicedFile(snippet, file) {
+    const staticFile = file;
+    let dynamicFile = file;
+    let fileLineNumber = 1;
+    let lookForStop = false;
+    let spliceStart = 0;
+    for (let [idx, _] of staticFile.lines.entries()) {
+      const line = file.lines[idx];
+      if (line.includes(writeStart)) {
+        const id = insertID(line);
+        if (id === snippet.id) {
+          spliceStart = fileLineNumber;
+          lookForStop = true;
+        }
+      }
+      if (line.includes(writeEnd) && lookForStop) {
+        dynamicFile = await this.spliceFile(
+          spliceStart,
+          fileLineNumber,
+          snippet,
+          dynamicFile
+        );
+        lookForStop = false;
+      }
+      fileLineNumber++;
+    }
+    return dynamicFile;
+  }
+
+  async spliceFile(start, end, snippet, file) {
+    const rmlines = end - start;
+    file.lines.splice(start, rmlines - 1, ...snippet.lines);
+    return file;
+  }
+
+  async clearSnippets(files) {
+    const clearProgress = new progress.Bar(
+      {
+        format: fmtProgressBar("starting clear operations"),
+        barsize: 20,
+      },
+      progress.Presets.shades_classic
+    );
+    clearProgress.start(files.length, 0);
+    for (let file of files) {
+      file = await this.getClearedFile(file);
+      clearProgress.increment();
+    }
+    clearProgress.stop();
+    return files;
+  }
+
+  async getClearedFile(file) {
+    let omit = false;
+    const newFileLines = [];
+    for (const line of file.lines) {
+      if (line.includes(writeEnd)) {
+        omit = false;
+      }
+      if (!omit) {
+        newFileLines.push(line);
+      }
+      if (line.includes(writeStart)) {
+        omit = true;
+      }
+    }
+    file.lines = newFileLines;
+    return file;
+  }
+
+  async writeFiles(files) {
+    const insertRootPath = join(rootDir, this.config.target);
+    const writeFileProgress = new progress.Bar(
+      {
+        format: fmtProgressBar("writing files to " + insertRootPath),
+        barsize: 20,
+      },
+      progress.Presets.shades_classic
+    );
+    writeFileProgress.start(files.length, 0);
+    for (const file of files) {
+      const fileString = `${file.lines.join("\n")}\n`;
+      const writePath = join(insertRootPath, file.filename);
+      await writeAsync(writePath, fileString);
+      writeFileProgress.increment();
+    }
+    writeFileProgress.stop();
+    return;
+  }
+
+  async cleanUp() {
+    const cleanupProgress = new progress.Bar(
+      {
+        format: fmtProgressBar("cleaning up downloads"),
+        barsize: 20,
+      },
+      progress.Presets.shades_classic
+    );
+    cleanupProgress.start(1, 0);
+    const path = join(rootDir, extractionDir);
+    rimrafAsync(path);
+    cleanupProgress.update(1);
+    cleanupProgress.stop();
+  }
+}
+
+function determineExtension(path) {
+    const parts = path.split(".");
+    return parts[parts.length - 1];
+  }
```

Comment:
[Sand] - Lets format these so there is no indentation for these functions.

Reply:
Changes made!